tests/profiler_test.py

```diff
@@ -205,5 +213,30 @@ def on_profile(port, logdir, worker_start):
     thread_worker.join(120)
     self._check_xspace_pb_exist(logdir)
 
+  @unittest.skipIf(
+      not (portpicker and profiler_client and tf_profiler and TBP_ENABLED),
+    "Test requires tensorflow.profiler, portpicker and "
+    "tensorboard_profile_plugin")
+  def test_remote_profiler(self):
+    port = portpicker.pick_unused_port()
+
+    ret = []
+    def on_profile():
+      with tempfile.TemporaryDirectory() as temp:
+        os.system(
+            f"python -m jax.collect_profile {port} 200 --log_dir {temp} "
+            "--no_perfetto_link")
+        ret.append(os.listdir(temp))
+
+    thread_profiler = threading.Thread(
+        target=on_profile, args=())
+    thread_profiler.start()
+    jax.profiler.start_server(port)
+    y = jnp.zeros((500, 500))
+    y = jnp.dot(y, y)
+    jax.profiler.stop_server()
+    thread_profiler.join()
+    self.assertTrue(len(ret[0]) > 0)
```

Comment:
Bonus points: check the profile contents. You should be able to reuse the L92-102

Reply:
I added a check for the presence of a "plugins" folder that would only be created if the profile plugin captured a trace.

Reply:
It's also good to check that the trace actually captured something, e.g. we used to have problems where we would generate a trace but without device or Python traces. That's what the test above does. Up to you though.

Reply:
I see, okay I used that helper function. Thanks!